### PR TITLE
Add ability to configure unsecure ports to the IPv6 datagram filter.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Comment by jwhui** [comment]
@darconeous

**Comment by mszczodrak** [comment]
LGTM

File: src/core/net/ip6_filter.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+ThreadError Filter::AddUnsecurePort(uint16_t aPort)
+{
+    ThreadError error = kThreadError_None;
+
+    for (int i = 0; i < kMaxUnsecurePorts; i++)
+    {
+        VerifyOrExit(aPort != mUnsecurePorts[i], error = kThreadError_Busy);
+    }
```
Comment (abtink) [inline src/core/net/ip6_filter.cpp]: Just a suggestion: If the added port is already on the list, we can say the `Add` was sort of a success and consider returning error `kThreadError_None` here.

**Comment by abtink** [comment]
Looks good.

**Comment by darconeous** [comment]
👍

**Pull request merged**